File: src/dbflute/adf/meta/vendorLargeDataRefdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorLargeDataRefDbm_T struct {
	df.BaseDBMeta
	ColumnLargeDataRefId *df.ColumnInfo
	ColumnLargeDataId *df.ColumnInfo
	ColumnDateIndex *df.ColumnInfo
	ColumnDateNoIndex *df.ColumnInfo
	ColumnTimestampIndex *df.ColumnInfo
	ColumnTimestampNoIndex *df.ColumnInfo
	ColumnNullableDecimalIndex *df.ColumnInfo
	ColumnNullableDecimalNoIndex *df.ColumnInfo
	ColumnSelfParentId *df.ColumnInfo
}

func (b *VendorLargeDataRefDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorLargeDataRefDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorLargeDataRefDbm *VendorLargeDataRefDbm_T

func Create_VendorLargeDataRefDbm() {
	VendorLargeDataRefDbm = new(VendorLargeDataRefDbm_T)
	VendorLargeDataRefDbm.TableDbName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TableDispName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TablePropertyName = "vendorLargeDataRef"
	VendorLargeDataRefDbm.TableSqlName = new(df.TableSqlName)
	VendorLargeDataRefDbm.TableSqlName.TableSqlName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TableSqlName.CorrespondingDbName = VendorLargeDataRefDbm.TableDbName

	var vendorLargeDataRef df.DBMeta
	vendorLargeDataRef = VendorLargeDataRefDbm
	VendorLargeDataRefDbm.DBMeta=&vendorLargeDataRef
	largeDataRefIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo LARGE_DATA_REF_ID
	largeDataRefIdSqlName.ColumnSqlName = "LARGE_DATA_REF_ID"
	largeDataRefIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnLargeDataRefId = df.CCI(&vendorLargeDataRef, "LARGE_DATA_REF_ID", largeDataRefIdSqlName, "", "", "Long.class", "largeDataRefId", "", true, false,true, "BIGINT", 19, 0, "",false,"","", "","","",false,"int64")
	largeDataIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo LARGE_DATA_ID
	largeDataIdSqlName.ColumnSqlName = "LARGE_DATA_ID"
	largeDataIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnLargeDataId = df.CCI(&vendorLargeDataRef, "LARGE_DATA_ID", largeDataIdSqlName, "", "", "Long.class", "largeDataId", "", false, false,true, "BIGINT", 19, 0, "",false,"","", "","","",false,"int64")
	dateIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo DATE_INDEX
	dateIndexSqlName.ColumnSqlName = "DATE_INDEX"
	dateIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnDateIndex = df.CCI(&vendorLargeDataRef, "DATE_INDEX", dateIndexSqlName, "", "", "java.time.LocalDate.class", "dateIndex", "", false, false,true, "DATE", 10, 0, "",false,"","", "","","",false,"df.MysqlDate")
	dateNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo DATE_NO_INDEX
	dateNoIndexSqlName.ColumnSqlName = "DATE_NO_INDEX"
	dateNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnDateNoIndex = df.CCI(&vendorLargeDataRef, "DATE_NO_INDEX", dateNoIndexSqlName, "", "", "java.time.LocalDate.class", "dateNoIndex", "", false, false,true, "DATE", 10, 0, "",false,"","", "","","",false,"df.MysqlDate")
	timestampIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo TIMESTAMP_INDEX
	timestampIndexSqlName.ColumnSqlName = "TIMESTAMP_INDEX"
	timestampIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnTimestampIndex = df.CCI(&vendorLargeDataRef, "TIMESTAMP_INDEX", timestampIndexSqlName, "", "", "java.time.LocalDateTime.class", "timestampIndex", "", false, false,true, "DATETIME", 19, 0, "",false,"","", "","","",false,"df.MysqlTimestamp")
	timestampNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo TIMESTAMP_NO_INDEX
	timestampNoIndexSqlName.ColumnSqlName = "TIMESTAMP_NO_INDEX"
	timestampNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnTimestampNoIndex = df.CCI(&vendorLargeDataRef, "TIMESTAMP_NO_INDEX", timestampNoIndexSqlName, "", "", "java.time.LocalDateTime.class", "timestampNoIndex", "", false, false,true, "DATETIME", 19, 0, "",false,"","", "","","",false,"df.MysqlTimestamp")
	nullableDecimalIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo NULLABLE_DECIMAL_INDEX
	nullableDecimalIndexSqlName.ColumnSqlName = "NULLABLE_DECIMAL_INDEX"
	nullableDecimalIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnNullableDecimalIndex = df.CCI(&vendorLargeDataRef, "NULLABLE_DECIMAL_INDEX", nullableDecimalIndexSqlName, "", "", "java.math.BigDecimal.class", "nullableDecimalIndex", "", false, false,false, "DECIMAL", 12, 3, "",false,"","", "","","",false,"df.NullNumeric")
	nullableDecimalNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo NULLABLE_DECIMAL_NO_INDEX
	nullableDecimalNoIndexSqlName.ColumnSqlName = "NULLABLE_DECIMAL_NO_INDEX"
	nullableDecimalNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnNullableDecimalNoIndex = df.CCI(&vendorLargeDataRef, "NULLABLE_DECIMAL_NO_INDEX", nullableDecimalNoIndexSqlName, "", "", "java.math.BigDecimal.class", "nullableDecimalNoIndex", "", false, false,false, "DECIMAL", 12, 3, "",false,"","", "","","",false,"df.NullNumeric")
	selfParentIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo SELF_PARENT_ID
	selfParentIdSqlName.ColumnSqlName = "SELF_PARENT_ID"
	selfParentIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnSelfParentId = df.CCI(&vendorLargeDataRef, "SELF_PARENT_ID", selfParentIdSqlName, "", "", "Long.class", "selfParentId", "", false, false,false, "BIGINT", 19, 0, "",false,"","", "","","",false,"sql.NullInt64")

	VendorLargeDataRefDbm.ColumnInfoList = new(df.List)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnLargeDataRefId)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnLargeDataId)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnDateIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnDateNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnTimestampIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnTimestampNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnNullableDecimalIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnNullableDecimalNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnSelfParentId)


	VendorLargeDataRefDbm.ColumnInfoMap=make(map[string]int)
	VendorLargeDataRefDbm.ColumnInfoMap["largeDataRefId"]=0
		VendorLargeDataRefDbm.ColumnInfoMap["largeDataId"]=1
		VendorLargeDataRefDbm.ColumnInfoMap["dateIndex"]=2
		VendorLargeDataRefDbm.ColumnInfoMap["dateNoIndex"]=3
		VendorLargeDataRefDbm.ColumnInfoMap["timestampIndex"]=4
		VendorLargeDataRefDbm.ColumnInfoMap["timestampNoIndex"]=5
		VendorLargeDataRefDbm.ColumnInfoMap["nullableDecimalIndex"]=6
		VendorLargeDataRefDbm.ColumnInfoMap["nullableDecimalNoIndex"]=7
		VendorLargeDataRefDbm.ColumnInfoMap["selfParentId"]=8
	    VendorLargeDataRefDbm.PrimaryKey = true
    VendorLargeDataRefDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorLargeDataRef
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorLargeDataRefDbm.ColumnLargeDataRefId)

	VendorLargeDataRefDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorLargeDataRefDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorLargeDataRefMeta df.DBMeta = VendorLargeDataRefDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorLargeDataRef"] = &vendorLargeDataRefMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorLargeDataRefDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_large_data_ref. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorLargeDataRefDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorLargeDataRefDbm _instance = new VendorLargeDataRefDbm();
    private VendorLargeDataRefDbm() {}
    public static VendorLargeDataRefDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getLargeDataRefId(), (et, vl) -> ((VendorLargeDataRef)et).setLargeDataRefId(ctl(vl)), "largeDataRefId");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getLargeDataId(), (et, vl) -> ((VendorLargeDataRef)et).setLargeDataId(ctl(vl)), "largeDataId");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getDateIndex(), (et, vl) -> ((VendorLargeDataRef)et).setDateIndex(ctld(vl)), "dateIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getDateNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setDateNoIndex(ctld(vl)), "dateNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getTimestampIndex(), (et, vl) -> ((VendorLargeDataRef)et).setTimestampIndex(ctldt(vl)), "timestampIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getTimestampNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setTimestampNoIndex(ctldt(vl)), "timestampNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getNullableDecimalIndex(), (et, vl) -> ((VendorLargeDataRef)et).setNullableDecimalIndex(ctb(vl)), "nullableDecimalIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getNullableDecimalNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setNullableDecimalNoIndex(ctb(vl)), "nullableDecimalNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getSelfParentId(), (et, vl) -> ((VendorLargeDataRef)et).setSelfParentId(ctl(vl)), "selfParentId");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_large_data_ref";
    protected final String _tableDispName = "vendor_large_data_ref";
    protected final String _tablePropertyName = "vendorLargeDataRef";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_large_data_ref", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnLargeDataRefId = cci("LARGE_DATA_REF_ID", "LARGE_DATA_REF_ID", null, null, Long.class, "largeDataRefId", null, true, false, true, "BIGINT", 19, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnLargeDataId = cci("LARGE_DATA_ID", "LARGE_DATA_ID", null, null, Long.class, "largeDataId", null, false, false, true, "BIGINT", 19, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnDateIndex = cci("DATE_INDEX", "DATE_INDEX", null, null, java.time.LocalDate.class, "dateIndex", null, false, false, true, "DATE", 10, 0, null, false, null, null, null, null, null, false);
//"df.MysqlDate"
    protected final ColumnInfo _columnDateNoIndex = cci("DATE_NO_INDEX", "DATE_NO_INDEX", null, null, java.time.LocalDate.class, "dateNoIndex", null, false, false, true, "DATE", 10, 0, null, false, null, null, null, null, null, false);
//"df.MysqlDate"
    protected final ColumnInfo _columnTimestampIndex = cci("TIMESTAMP_INDEX", "TIMESTAMP_INDEX", null, null, java.time.LocalDateTime.class, "timestampIndex", null, false, false, true, "DATETIME", 19, 0, null, false, null, null, null, null, null, false);
//"df.MysqlTimestamp"
    protected final ColumnInfo _columnTimestampNoIndex = cci("TIMESTAMP_NO_INDEX", "TIMESTAMP_NO_INDEX", null, null, java.time.LocalDateTime.class, "timestampNoIndex", null, false, false, true, "DATETIME", 19, 0, null, false, null, null, null, null, null, false);
//"df.MysqlTimestamp"
    protected final ColumnInfo _columnNullableDecimalIndex = cci("NULLABLE_DECIMAL_INDEX", "NULLABLE_DECIMAL_INDEX", null, null, java.math.BigDecimal.class, "nullableDecimalIndex", null, false, false, false, "DECIMAL", 12, 3, null, false, null, null, null, null, null, false);
//"df.NullNumeric"
    protected final ColumnInfo _columnNullableDecimalNoIndex = cci("NULLABLE_DECIMAL_NO_INDEX", "NULLABLE_DECIMAL_NO_INDEX", null, null, java.math.BigDecimal.class, "nullableDecimalNoIndex", null, false, false, false, "DECIMAL", 12, 3, null, false, null, null, null, null, null, false);
//"df.NullNumeric"
    protected final ColumnInfo _columnSelfParentId = cci("SELF_PARENT_ID", "SELF_PARENT_ID", null, null, Long.class, "selfParentId", null, false, false, false, "BIGINT", 19, 0, null, false, null, null, null, null, null, false);
//"sql.NullInt64"

    /**
     * LARGE_DATA_REF_ID: {PK, NotNull, BIGINT(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnLargeDataRefId() { return _columnLargeDataRefId; }
    /**
     * LARGE_DATA_ID: {NotNull, BIGINT(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnLargeDataId() { return _columnLargeDataId; }
    /**
     * DATE_INDEX: {NotNull, DATE(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDateIndex() { return _columnDateIndex; }
    /**
     * DATE_NO_INDEX: {NotNull, DATE(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDateNoIndex() { return _columnDateNoIndex; }
    /**
     * TIMESTAMP_INDEX: {NotNull, DATETIME(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampIndex() { return _columnTimestampIndex; }
    /**
     * TIMESTAMP_NO_INDEX: {NotNull, DATETIME(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampNoIndex() { return _columnTimestampNoIndex; }
    /**
     * NULLABLE_DECIMAL_INDEX: {DECIMAL(12, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNullableDecimalIndex() { return _columnNullableDecimalIndex; }
    /**
     * NULLABLE_DECIMAL_NO_INDEX: {DECIMAL(12, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNullableDecimalNoIndex() { return _columnNullableDecimalNoIndex; }
    /**
     * SELF_PARENT_ID: {BIGINT(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnSelfParentId() { return _columnSelfParentId; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnLargeDataRefId());
        ls.add(columnLargeDataId());
        ls.add(columnDateIndex());
        ls.add(columnDateNoIndex());
        ls.add(columnTimestampIndex());
        ls.add(columnTimestampNoIndex());
        ls.add(columnNullableDecimalIndex());
        ls.add(columnNullableDecimalNoIndex());
        ls.add(columnSelfParentId());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnLargeDataRefId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorLargeDataRef"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorLargeDataRefCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorLargeDataRefBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorLargeDataRef> getEntityType() { return VendorLargeDataRef.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorLargeDataRef newEntity() { return new VendorLargeDataRef(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorLargeDataRef)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorLargeDataRef)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
